lib/dead_end/core_ext.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+# Allow lazy loading, only load code if/when there's a syntax error
+autoload :DeadEnd, "dead_end/api"
+
+# Ruby 3.2+ has a cleaner way to hook into Ruby that doesn't use `require`
+if SyntaxError.new.respond_to?(:detailed_message)
+  module DeadEndUnloaded
```

Comment:
Why isn't this `UnloadedDeadEnd` in dead_end/version.rb?

Reply:
When developing the lazy load feature this constant came first, the constant in the version.rb was only needed when the gemspec gets loaded and was not discovered until later. I never went back and updated this file to depend on the declaration in version.rb but I can make that update.